spec/models/response_csv_spec.rb

```diff
@@ -63,8 +63,68 @@
 
     it "should generate correct CSV" do
       responses = Response.unscoped.with_associations.order(:created_at)
-      expected = File.read(File.expand_path('../../expectations/responses.csv', __FILE__))
+      expected = File.read(File.expand_path('../../expectations/response_csv/responses.csv', __FILE__))
       expect(ResponseCSV.new(responses).to_s).to eq expected
     end
   end
-end
\ No newline at end of file
+
+  context "with repeat groups" do
+    let(:repeat_form) do
+      create(:form, question_types: ["integer", [:repeating, "text", "integer", "select_multiple"], "integer", [:repeating, "text", "geo_multilevel_select_one",  "integer"]]).tap do |f|
```

Comment:
this is interesting, but it looks so much like a hash that i'm wondering if this would be a better API for this:

`create(:form, question_types: ["integer", {repeating: ["text", "integer", "select_multiple"]}, "integer", {repeating: ["text", "geo_multilevel_select_one",  "integer"]}])`

Benefits include that you can just do a simple `hash[:repeating]` to get the question types rather than trying to delete the magic from the array. I think it also delineates the boundary a little better. This is a really good way of eliminating our dependency on plucking the specific group manually out of the code by its position in the array to apply the transformation on its attributes.